core-bundle/src/Search/Document.php

```diff
@@ -145,8 +146,23 @@ static function (Crawler $node) {
             )
         ;
 
-        // Filter invalid (null) values
-        $this->jsonLds = array_filter($this->jsonLds);
+        // Filter invalid (null) and parse all values
+        foreach (array_filter($jsonLds) as $jsonLd) {
+            // If array has numeric keys, it likely contains multiple data inside it which should be
+            // treated as if coming from separate sources, and thus moved to the root of an array.
+            $jsonLdItems = ArrayUtil::isAssoc($jsonLd) ? [$jsonLd] : $jsonLd;
+
+            // Parsed the grouped values under the @graph within the same context
+            foreach ($jsonLdItems as $jsonLdItem) {
+                if (isset($jsonLdItem['@graph']) && \is_array($jsonLdItem['@graph'])) {
+                    foreach ($jsonLdItem['@graph'] as $graph) {
+                        $this->jsonLds[] = array_merge(['@context' => $jsonLdItem['@context']], $graph);
```

Comment:
Can we be sure the `@context` key exists?

Reply:
No we can't, we need to treat it as optional.

```json
{
    "@context": "https://schema.org",
    "name": "foobar"
}
```

is the same as


```json
{
    "https://schema.org/name": "foobar"
}
```

We already handle it in `expandJsonLdContexts()` so this should be re-used here, you're right!

Reply:
I changed the logic here in 7ce0abc7 to not rely purely on `@context` but rather on anything that's there, so 

```
{
    "https://schema.org/name": "foobar",
    "@graph" => […],
}
```

is handled the same way as with `@context` thingy.